src/components/Global/ImageScaler.vue

```diff
@@ -0,0 +1,150 @@
+<script setup>
+import { ref, computed, onMounted, defineEmits, defineProps } from 'vue'
+import { useConfigurationStore } from '@/stores/configuration'
+import { fetchApiCall } from '@/utils/api'
+import RawScaledImage from './RawScaledImage.vue'
+
+// This component gets the raw image data from the server for an image
+// and then displays the image and the scaling controls under it
+
+const props = defineProps({
+  image: {
+    type: Object,
+    required: true
+  },
+  maxSize: {
+    type: Number,
+    default: 500
+  },
+  imageName: {
+    type: String,
+    required: true
+  },
+  compositeName: {
+    type: String,
+    required: true
+  }
+})
+
+const emit = defineEmits(['updateScaling'])
+const store = useConfigurationStore()
+const dataSessionsUrl = store.datalabApiBaseUrl
+const isLoading = ref(true)
+const errorReason = ref('')
+const rawData = ref({})
+const sliderRange = ref([0, 65535])
+var zScaleValues = null
+
+const titleText = computed(() => {
+  return props.imageName.replace('_', ' ')
+})
+
+function updateLowerScale(value) {
+  sliderRange.value = [Number(value), sliderRange.value[1]]
+  emit('updateScaling', props.imageName, sliderRange.value[0], sliderRange.value[1])
+}
+
+function updateUpperScale(value) {
+  sliderRange.value = [sliderRange.value[0], Number(value)]
+  emit('updateScaling', props.imageName, sliderRange.value[0], sliderRange.value[1])
+}
+
+function zScaleImage() {
+  if (zScaleValues){
+    sliderRange.value = [...zScaleValues]
+  }
+  emit('updateScaling', props.imageName, sliderRange.value[0], sliderRange.value[1])
+}
+
+
+onMounted(async () => {
+  const url = dataSessionsUrl + 'analysis/raw-data/'
+  const body = {
+    'basename': props.image.basename,
+    'source': props.image.source,
+    'max_size': props.maxSize
+  }
+  fetchApiCall({url: url, method: 'POST', body: body,
+    successCallback: (response) => {
+      rawData.value = response
+      zScaleValues = [response.zmin, response.zmax]
+      zScaleImage()
+      isLoading.value = false
+    },
+    failCallback: (error) => {
+      isLoading.value = false
+      errorReason.value = error
+      console.error('API call failed with error:', error)
+    }
```

Comment:
this is a very readable way to use lamdas for the fetchapi method, I'll be stealing this